lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -466,10 +466,16 @@ namespace {
 /// spelling of the protocol type, as well as the locality in the file), but it
 /// does work.
 class InheritedProtocolCollector {
+  static const constexpr StringLiteral DummyProtocolName =
+      "_ConstraintThatIsNotPartOfTheAPIOfThisLibrary";
+
   /// Protocols that will be included by the ASTPrinter without any extra work.
   SmallVector<ProtocolDecl *, 8> IncludedProtocols;
   /// Protocols that will not be printed by the ASTPrinter.
   SmallVector<ProtocolDecl *, 8> ExtraProtocols;
+  /// Protocols that can be printed, but whose conformances are constrained with
+  /// something that \e can't be printed.
+  SmallVector<const ProtocolType *, 8> ConditionalConformanceProtocols;
```

Comment:
Planck-length-sized-nit: Can you rename this to include the fact that these won't be printed?